analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+         *          |                                         |
+         *          |                                Block6<--+
+         *          |                              #2 is null
+         *          |                               /   \
+         *          |                             T/     \F
+         *          |                             /       \
+         *          |                           Block8   Block7
+         *          |                         #1="Hello"  #1=#2"
+         *          |                             |         |
+         *          +------------>Block9<---------+---------+
+         *                         s=#1
+         *                          |
+         *                         Exit
+         */
+        const string code = """string s = s1 ?? (s2 is null ? s3 : "NestedFalse") ?? "Hello";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1, string s2, string s3");
+        context.ValidateEntry(LiveIn("s1", "s2", "s3"), LiveOut("s1", "s2", "s3"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1", "s2", "s3"), LiveOut("s1", "s2", "s3"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("s1", "s2", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[7], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[8], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[9], null, LiveIn("s1", "s3"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_Overwrite()
+    {
+        /*       Block 1
+         *       #0=s1
+         *         |
+         *       Block 2
+         *   #1=(s1="overwrite")
+         *     #1 is null
+         *        /   \
+         *       T     F
+         *       /     \
+         *    Block3   Block4
+         *    #2=#1   #2="value"
+         *      \         /
+         *       \       /
+         *        Block5
+         *        s1=#2
+         */
+        const string code = """s1 = (s1 = "overwrite")  ?? "value";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1");
+        context.ValidateEntry(LiveIn("s1"), LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1"));
+        context.Validate(context.Cfg.Blocks[2], null);
+        context.Validate(context.Cfg.Blocks[3], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveOut("s1"));
```

Comment:
Block 2, 3, and 4 are suspicious. It might be correct, but it deserves a comment with an explanation at least
Why doesn't have block 3 a `LiveIn("s1")`

Reply:
Capture-1 gets as value the full assignment operation and it does not resolve to a symbol later.
Maybe this is a ticket for more hardening. WDYT?

Reply:
The comment is enough. I would create issue only if you can craft a FP from it.

Reply:
Please add also comment that there should be LiveIn in block 3 and 4, to make our future life a bit easier once we bump into it int he future - with no explanation, it's already on block 2